Archipelago.HollowKnight/HintTracker.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Archipelago.HollowKnight.IC;
+using Archipelago.MultiClient.Net;
+using Archipelago.MultiClient.Net.Models;
+using ItemChanger;
+using ItemChanger.Internal;
+
+namespace Archipelago.HollowKnight;
+
+public static class HintTracker
+{
+
+    public static event Action OnArchipelagoHintUpdate;
+    
+    /// <summary>
+    ///  List of MultiClient.Net Hint's
+    /// </summary>
+    public static List<Hint> Hints;
+
+    private static ArchipelagoSession _session;
+
+    private static void UpdateHints(Hint[] arrayHints)
+    {
+        Hints = arrayHints.ToList();
+        foreach (var hint in Hints)
+        {
+            if (hint.FindingPlayer != Archipelago.Instance.session.ConnectionInfo.Slot)
+                continue;
+
+            var locationName = StripShopSuffix(_session.Locations.GetLocationNameFromId(hint.LocationId));
+            var location = Finder.GetLocation(locationName);
+            if (location == null)
+                continue;
+            if (!Ref.Settings.Placements.ContainsKey(locationName))
+                continue;
+
+            var placement = Ref.Settings.Placements[locationName];
```

Comment:
I think we can make use of `Archipelago.Instance.placementsByLocationID` here instead

Reply:
placementsByLocationID is empty upon resuming a save file at this point in the code.
that is not populated until all tags get loaded. 
is there an event that fires after all tags/itemtags get loaded by IC?

Reply:
Events.OnEnterGame should work. This makes me wonder if StartOrResumeGame is actually executing at the correct time. I think ModHooks.SaveGameLoadHook should definitely be replaced by this event. I am also a little worried about when some of these things are executing on new saves as well. Having interactions with the server start before placements are loaded seems extremely dangerous to me

Reply:
This might be tech debt for later though

Reply:
On second thought leave it as-is and I'll save this cleanup for #159

Reply:
Wait, I missed that you said this is only an issue on resume, so try changing that event (where `startorresumegame(false) is called` and if that works for both new and resumed files we can fix it here final answer I swear